R/zchunk_LB172.nonghg_elc_USA.R

```diff
@@ -0,0 +1,165 @@
+      # Subset electricity emissions
+      filter(GCAM_sector == "elec_heat") %>%
+      # GCAM fuel, missing value drop later
+      left_join(CEDS_GCAM_fuel, by = "CEDS_Fuel") %>%
+      rename(fuel = GCAM_fuel, NEI_pollutant = pollutant) %>%
+      # Match on NEI pollutants, using left_join becuase missing values will be produced and dropped later
+      left_join(NEI_pollutant_mapping, by = "NEI_pollutant") %>%
+      # MISSING VALUES: PM filterable. Not needed bc have filt+cond. OK to omit
+      na.omit %>%
+      # Convert from short ton to Tg
+      mutate(emissions = emissions / CONV_T_METRIC_SHORT / 10 ^ 6, unit = "Tg") %>%
+      # Organize
+      rename(sector = GCAM_sector) %>%
+      group_by(state, sector, fuel, Non.CO2) %>%
+      summarise(emissions = sum(emissions)) %>%
+      rename(value = emissions) %>%
+      mutate(year = 2010) %>%
+      ungroup
+
+    ## Add specific Non.CO2 species for NOx_Coal, SO2_Coal, NOx_ELEC, SO2_ELEC for the base year
+    # These species will be used in CSAPR constrains in policy files
+    # 1) Add N0x_Coal
+    L172.NOx_Coal_Yb <- L172.nonghg_tg_state_elec_F_Yb %>%
+      filter(fuel == "coal" & Non.CO2 == "NOx") %>%
+      mutate(Non.CO2 = "NOx_Coal")
+    # 2) Add S02_Coal
+    L172.SO2_Coal_Yb <- L172.nonghg_tg_state_elec_F_Yb %>%
+      filter(fuel == "coal" & Non.CO2 == "SO2") %>%
+      mutate(Non.CO2 = "SO2_Coal")
+    # 3) Add NOx_ELEC
+    L172.NOx_ELEC_Yb <- L172.nonghg_tg_state_elec_F_Yb %>%
+      filter(Non.CO2 == "NOx") %>%
+      mutate(Non.CO2 = "NOx_ELEC")
+    # 4) Add SO2_ELEC
+    L172.SO2_ELEC_Yb <- L172.nonghg_tg_state_elec_F_Yb %>%
+      filter(Non.CO2 == "SO2") %>%
+      mutate(Non.CO2 = "SO2_ELEC")
+    # Add these sudo species into L172 main data
+    L172.nonghg_tg_state_elec_F_Yb <- bind_rows(L172.nonghg_tg_state_elec_F_Yb,
+                                                L172.NOx_Coal_Yb, L172.SO2_Coal_Yb,
+                                                L172.NOx_ELEC_Yb, L172.SO2_ELEC_Yb)
+
+
+    ## Input-based emissions in the future years, use data from EPA-ORD
+    # L172.nonghg_tgej_state_elec_F_Yf: Electricity non-co2 emissions coefficients by fuel input and U.S. state in future model years
+
+    L172.nonghg_tgej_state_elec_F_Yf_prior2025 <- EPA_state_egu_emission_factors_ktPJ %>%
+      # Convert to long format
+      gather(variable, value, -state_name, -fuel) %>%
+      separate(variable, into = c("year","Non.CO2"), sep = "_") %>%
+      mutate(year = as.numeric(year)) %>%
+      # NOTE: for now change oil to refined liquids
+      mutate(fuel = gsub("oil","refined liquids", fuel)) %>%
+      # state code & select relevant columns
+      left_join_error_no_match(states_subregions %>% select(state, state_name),
+                by = c("state_name")) %>%
+      mutate(sector = "elec_heat") %>%
+      select(state, sector, fuel, Non.CO2, year, value) %>%
+      # filter only future model years prior to 2025
+      filter(year %in% FUTURE_YEARS[1:3])
```

Comment:
This (and lines 112 and 114 below) is super fragile and will quickly break if `FUTURE_YEARS` is redefined. Let's find a better way to do this, e.g. use a `INPUT_EMISSION_YEARS` constant.